src/clients/c/tb_client.h

```diff
@@ -230,7 +230,7 @@ typedef enum TB_QUERY_FILTER_FLAGS {
 // This struct must be "pinned" (not copyable or movable), as its address must remain stable
 // throughout the lifetime of the client instance.
 typedef struct tb_client_t {
-    uint64_t opaque[4];
+    uint64_t opaque[3];
```

Comment:
I'd maybe keep it as `4`, to have some extra bytes, and to be a power-of-two

Reply:
I'm wondering ... could we use this extra byte to somehow validate the struct? Like a hash or a magic number, so we return `CLIENT_INVALID` instead of SEGFAULT'ng if the user pass a garbage pointer.

Reply:
Brilliant! Could write `0xBEE71E` there and assert! 

We should also write it at the _end_ of `init`, so that, if init bails part-way through, but the user doesn't handle the error and tries to use it, we'll still assert!